httplogger/httplogger.go

```diff
@@ -108,36 +111,43 @@ func LoggingMiddlewareLogrus(logger *logrus.Entry, next http.Handler) http.Handl
 }
 
 // LoggingMiddlewareZap logs the incoming HTTP request & its duration.
-func LoggingMiddlewareZap(logger *zap.SugaredLogger, next http.Handler) http.Handler {
-	return http.HandlerFunc(
-		func(w http.ResponseWriter, r *http.Request) {
-			defer func() {
-				if err := recover(); err != nil {
-					w.WriteHeader(http.StatusInternalServerError)
-
-					method := ""
-					url := ""
-					if r != nil {
-						method = r.Method
-						url = r.URL.EscapedPath()
-					}
-
-					logger.With(
-						"err", err,
-						"trace", string(debug.Stack()),
-						"method", r.Method,
-					).Errorf("http request panic: %s %s", method, url)
+func LoggingMiddlewareZap(logger *zap.Logger, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Generate request ID (`base64` to shorten its string representation)
+		_uuid := [16]byte(uuid.New())
+		httpRequestID := base64.RawStdEncoding.EncodeToString(_uuid[:])
+
+		l := logger.With(zap.String("httpRequestID", httpRequestID))
+		r = logutils.RequestWithZap(r, l)
+
+		// Handle panics
+		defer func() {
+			if msg := recover(); msg != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				var method, url string
+				if r != nil {
+					method = r.Method
+					url = r.URL.EscapedPath()
 				}
-			}()
-			start := time.Now()
-			wrapped := wrapResponseWriter(w)
-			next.ServeHTTP(wrapped, r)
-			logger.With(
-				"status", wrapped.status,
-				"method", r.Method,
-				"path", r.URL.EscapedPath(),
-				"duration", fmt.Sprintf("%f", time.Since(start).Seconds()),
-			).Infof("http: %s %s %d", r.Method, r.URL.EscapedPath(), wrapped.status)
-		},
-	)
+				l.Error("HTTP request handler panicked",
+					zap.Any("error", msg),
+					zap.String("method", method),
+					zap.String("url", url),
+				)
+			}
+		}()
+
+		start := time.Now()
+		wrapped := wrapResponseWriter(w)
+		next.ServeHTTP(w, r)
+
+		// Passing request stats both in-message (for the human reader)
+		// as well as inside the structured log (for the machine parser)
+		l.Info(fmt.Sprintf("Handled HTTP request: %s %s %d", r.Method, r.URL.EscapedPath(), wrapped.status),
+			zap.Duration("duration", time.Since(start)),
```

Comment:
let's add the unit, and log duration milliseconds

```suggestion
			zap.Duration("durationMs", time.Since(start).Milliseconds()),
```

Reply:
`zap.Duration` would log as "55µs", but we want a specific unit for metrics, always. therefore it's better to be explicit here, and make sure milliseconds is logged